### ISSUE TITLE
Allow to use back-key as ESC

### ISSUE THREAD

**Issue by pvxx** [vx/connectbot#17]
VX Connectbot is really great, as a heavy VIM user I miss only one feature I like in Android Terminal Emulator, using back-key as ESC. I have Sony Xperia Pro (MK16i) and globe-key is not very convenient for this, especially if you have to press it two times.

If you could add this as an option, I think all VIM users will be grateful :).

It would be nice to utilize volume up/down keys as well (for example for F-n keys), if possible.

Of course the best way would be to allow custom keyboard profile stored in some config-file on sd-card :), but maybe it would complicate the code too much to be practical.

**Comment by zasran** [comment]
Yes, ESC is crucial, for vim or midnight commander etc. Would really like to see ability to re-map keyboard using a config file (or menu).

**Comment by reybalgs** [comment]
I totally agree with this. I use an ASUS Eee Pad Transformer TF101 and I get frustrated when I press the back key on my keyboard dock (since that's where the ESC is located on standard keyboards) and the SSH terminal session goes out and I'm put back in the main menu. I also regularly use Vim as my editor while I'm connected to my remote machine, so having the hardware back key mapped to ESC would be awesome. Terminal IDE has a toggle option for this, so implementing this shouldn't be too hard.

**Comment by anerdwithaknife** [comment]
I agree, right now I'm torn between Better Terminal Emulator Pro that handles the keyboard on my TF101 perfectly but has no support for UTF-8 and VX Connectbot with excellent support for everything except mapping the ESC key to back button on the hardware keyboard. 

(I'm a heavy Vim-user, but also use ESC for meta in irssi and other programs.)

**Comment by ghost** [comment]
I totally agree. This feature is a must for view users. I use a Bluetooth keyboard that has android keys and in place of escape is back.  So for vi all the flavors of connectbot I've used so far are extremely crippled for me.  I use the same keyboard though on iPad with connectbot there and the SAME keyboard works perfect for esc

**Comment by OliverUv** [comment]
I also request this feature.

**Comment by o2genum** [comment]
Yes, we want it to be possible to remap some key to ESC.

**Comment by metasoarous** [comment]
+1. This is a pretty vital feature for me. Not having it basically makes this app unusable for me.

**Comment by pabranch** [comment]
+1 Vital for me as well.

**Comment by quezak** [comment]
+1, it would be useful

**Comment by ghost** [comment]
Any update on this? Just ran into the same issue... vim is unusable.
